### PR TITLE
python3.pkgs.keyring: 21.5.0 -> 23.0.0

### PR DESCRIPTION
**Pull request by dotlambda**
depends on https://github.com/NixOS/nixpkgs/pull/113028

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/keyring/default.nix [hunk 1]
```diff
@@ -1,43 +1,47 @@
...
-  checkInputs = [ pytest pytest-flake8 ];
+  checkInputs = [
+    pytestCheckHook
+  ];
 
-  propagatedBuildInputs = [ dbus-python entrypoints ]
-  ++ lib.optional stdenv.isLinux secretstorage
-  ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ];
+  propagatedBuildInputs = [
+    importlib-metadata
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/keyring/default.nix:31]: The pythonOlder got lost.

File: pkgs/development/python-modules/keyring/default.nix [hunk 2]
```diff
@@ -1,43 +1,47 @@
...
   disabled = isPy27;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "207bd66f2a9881c835dad653da04e196c678bf104f8252141d2d3c4f31051579";
+    sha256 = "237ff44888ba9b3918a7dcb55c8f1db909c95b6f071bfb46c6918f33f453a68a";
   };
 
   nativeBuildInputs = [
     setuptools_scm
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/keyring/default.nix:22]:
```suggestion
    setuptools-scm
```

File: pkgs/development/python-modules/keyring/default.nix [hunk 3]
```diff
@@ -1,43 +1,47 @@
 { lib, stdenv, buildPythonPackage, fetchPypi, isPy27, pythonOlder
-, dbus-python
-, entrypoints
-, importlib-metadata
-, pytest
-, pytest-flake8
-, secretstorage
 , setuptools_scm
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/keyring/default.nix:2]:
```suggestion
, setuptools-scm
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/keyring/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/keyring/default.nix:31]: https://github.com/jaraco/keyring/blob/v23.0.0/setup.cfg#L28

**Comment by dotlambda** [comment]
included in https://github.com/NixOS/nixpkgs/pull/117452

File: pkgs/development/python-modules/keyring/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/keyring/default.nix:31]: The Package is integrated in 3.8 so upstream should correct their requires.

**Comment by SuperSandro2000** [comment]
> included in #117452

The updates in that PR are of lower quality because they often miss other changes except the update and to get it to build. Individual PRs are always preferred.

**Pull request closed without merging**